integrations/webchat/examples/download-history/client/javascript/index.html

```diff
@@ -25,34 +28,67 @@
       messages.push(event.data);
     }
 
+    /**
+     * This function will save each message that is generated by the send and receive events with a human agent.
+     */
+    function saveAgentMessage(event) {
+      agentMessages.push({
+        ...event.data,
+        // Add the agent's nickname to this object to make it easier to process both the bot and agent messages.
+        agentNickname: (event.agentProfile && event.agentProfile.nickname) || 'Agent',
+      });
+    }
+
     /**
      * This function will be called when the user selects the "Download history" custom menu option.
      */
     function createDownload() {
+      let downloadLines = [];
+      downloadLines.push('Assistant messages')
+      downloadLines.push(createDownloadText('From', 'Message'));
+
+      downloadLines = downloadLines.concat(messagesToLines(messages));
+
+      if (agentMessages.length) {
```

Comment:
Whenever we combine the chat windows I would love for us to concat these lists into a single thread by timestamp instead of having the bot and human conversations be separate.